src/snowflake/snowpark/types.py

```diff
@@ -115,6 +121,11 @@ def __hash__(self):
         if self.length == StringType._MAX_LENGTH:
             return StringType().__hash__()
         return super().__hash__()
+        
+    def fill_ast(self, ast: proto.SpDataType) -> None:
+        # TODO: Create SpStringType IR entity with field "length: Int", and update
```

Comment:
instead of all the TODOs as comment, could also use NotImplementedError with a comment on what to do. You could also define an error message

`ErrorMessage.missing_ir('SpStringType IR needs field "length: Int"')` 

that would help a lot!

Reply:
I agree, raising an error on the client side would probably be the best approach to raise the alarm on any Snowpark API elements with no representation in the IR.

I was afraid of blocking the client from working while we evaluate the IR changes, but I can turn the `TODO`s into `NotImplementedError` if that is okay.

Please see the pull request I created addressing all of the `TODO`s here: https://github.com/snowflakedb/snowflake/pull/183083